### PR TITLE
 Added `protocolProfileBehavior` property in Item

### PR DESCRIPTION
**Pull request by codenirvana**
Closes #707

### CODE REVIEW

File: lib/collection/item.js [hunk 1]
```diff
@@ -8,6 +8,16 @@ var _ = require('../util').lodash,
...
 
+    /**
+     * Whitelisted protocol profile behaviors.
+     *
+     * @private
+     * @type {String}
+     *
+     * @todo change type to array for supporting multiple behaviors
+     */
+    PROTOCOL_PROFILE_BEHAVIORS = 'disableBodyPruning',
```
Comment (kunagpal) [inline lib/collection/item.js]: Array.

File: lib/collection/item.js [hunk 2]
```diff
@@ -134,7 +144,18 @@ _.inherit((
...
+
+            /**
+             * Set of configurations used to alter the usual behavior of sending the request.
+             *
+             * @type {Object}
+             * @property {Boolean} disableBodyPruning Disable body pruning for request methods like GET, HEAD etc.
+             *
+             * @todo Inherit this from ItemGroup & Collection
+             */
+            protocolProfileBehavior: definition && definition.protocolProfileBehavior &&
```
Comment (kunagpal) [inline lib/collection/item.js]: Ternary expression with `undefined` as a fallback.

File: lib/collection/item.js [hunk 3]
```diff
@@ -134,7 +142,18 @@ _.inherit((
...
```
Comment (shamasis) [inline lib/collection/item.js]: can we make this free form?

File: test/unit/item.test.js [hunk 4]
```diff
@@ -91,6 +94,12 @@ describe('Item', function () {
                     expect(item.responses.all()).to.be.an('array');
                     expect(item.responses.all()).to.not.be.empty();
                 });
+
+                it('protocolProfileBehavior', function () {
```
Comment (kunagpal) [inline test/unit/item.test.js]: Add a test for non-object protocolProfileBehaviour as well.

File: lib/collection/item.js [hunk 5]
```diff
@@ -134,7 +134,18 @@ _.inherit((
...
+
+            /**
+             * Set of configurations used to alter the usual behavior of sending the request.
+             *
+             * @type {Object}
+             * @property {Boolean} disableBodyPruning Disable body pruning for request methods like GET, HEAD etc.
+             *
+             * @todo Inherit this from ItemGroup & Collection
+             */
+            protocolProfileBehavior: definition && typeof definition.protocolProfileBehavior === 'object' ?
```
Comment (kunagpal) [inline lib/collection/item.js]: Shift `'object'` to a top-level `var`

**Pull request merged**